dashboard/test/models/levels/blockly_test.rb

```diff
@@ -1238,18 +1238,55 @@ def create_custom_block(name, pool, block_text, args, category: 'Events')
     assert_equal parsed_xml.at_xpath('//block[@type="studio_ask"]/*[@name="VAR"]').content, localized_variable_str
   end
 
-  test 'keeps tags expanded when localizing start_html' do
+  test 'localizes start_libraries' do
+    test_locale = 'te-ST'
+    level_name = 'test localize start_libraries'
+    i18n_library_name = 'i18n_library_test'
     level = create(
       :level,
       :blockly,
-      name: 'test localized_start_html',
+      name: level_name,
+      )
+
+    # Create a start_libraries blockly level JSON structure containing the
+    # a translatable library and a non-translatable library.
+    start_libraries = JSON.generate(
+      [{name: i18n_library_name,
+        description: "Test Library that gets translated",
+        source: "var TRANSLATIONTEXT = {\n \"test_key_1\": \"expected_string_1\",\"test_key_2\": \"expected_string_2\"\n};\n\n//This library is translated\nfunction someFunction(key) {\n return TRANSLATIONTEXT[key];\n}\n"},
+       {name: "non_translated_library",
+        description: "Test Library that does not get translated",
+        source: "var TRANSLATIONTEXT = {\n \"not_translated_key\": \"not_translated_string\"\n};\n\n//This library is not translated\nfunction someFunction(key) {\n return TRANSLATIONTEXT[key];\n}\n"}]
     )
-    start_html = '<div><button id="leftBottomButton"></button><label>Outfit Picker</label></div>'
-    localized_start_html = level.localized_start_html(start_html)
 
-    # Output should use <button></button> instead of <button />
-    expected_output = '<div><button id="leftBottomButton"></button><label>Outfit Picker</label></div>'
-    assert_equal expected_output, localized_start_html
+    # Add translation mapping to the I18n backend
+    custom_i18n = {
+      'data' => {
+        'start_libraries' => {
+          level_name => {
+            i18n_library_name => {
+              test_key_1: 'localized_string_1'
+            }
+          }
+        }
+      }
+    }
+    I18n.locale = test_locale
+    I18n.backend.store_translations test_locale, custom_i18n
+
+    # Localized output to be tested
+    localized_start_libraries = level.localized_start_libraries(start_libraries)
+
+    # Expected output
+    localized_library = JSON.generate(
+      [{name: i18n_library_name,
+        description: "Test Library that gets translated",
+        source: "var TRANSLATIONTEXT = {\n \"test_key_1\": \"localized_string_1\",\"test_key_2\": \"expected_string_2\"\n};\n\n//This library is translated\nfunction someFunction(key) {\n return TRANSLATIONTEXT[key];\n}\n"},
```

Comment:
This test currently fails because we `JSON.parse` and `JSON.generate` the object containing the `TRANSLATIONTEXT`, which removes escape characters like `\n`